src/chain/store/index.rs

```diff
@@ -151,139 +63,189 @@ impl<DB: Blockstore> ChainIndex<DB> {
         Ok(ts)
     }
 
-    /// Loads tipset at `to` [`ChainEpoch`], loading from sparse cache and/or
-    /// loading parents from the `blockstore`.
-    pub(in crate::chain) fn get_tipset_by_height(
+    /// Find tipset at epoch `to` in the chain of ancestors starting at `from`.
+    /// If the tipset is _not_ in the chain of ancestors (i.e., if the `to`
+    /// epoch is higher than `from.epoch()`), an error will be returned.
+    ///
+    /// # Why pass in the `from` argument?
+    ///
+    /// Imagine the database contains five tipsets and a genesis block in this
+    /// configuration:
+    ///
+    /// ```text
+    ///           ┌───────┐  ┌────────┐  ┌────────┐
+    /// Genesis◄──┤Epoch 1◄──┤Epoch 2A◄──┤Epoch 3A│
+    ///           └───▲───┘  └────────┘  └────────┘
+    ///               │      ┌────────┐  ┌────────┐
+    ///               └──────┤Epoch 2B◄──┤Epoch 3B│
+    ///                      └────────┘  └────────┘
+    /// ```
+    ///
+    /// Here we have a fork in the chain and it is ambiguous which tipset to
+    /// load when epoch 2 is requested. The ambiguity is solved by passing in a
+    /// younger tipset (higher epoch) from which has the desired tipset as an
+    /// ancestor.
+    /// Calling `get_tipset_by_height(2, epoch_3a)` will return `Epoch 2A`.
+    /// Calling `get_tipset_by_height(2, epoch_3b)` will return `Epoch 2B`.
+    ///
+    /// # What happens when a null tipset is requested?
+    ///
+    /// ```text
+    ///           ┌───────┐          ┌───────┐  ┌───────┐
+    /// Genesis◄──┤Epoch 1│   Null   │Epoch 3◄──┤Epoch 4│
+    ///           └───▲───┘          └───┬───┘  └───────┘
+    ///               │                  │
+    ///               └──────────────────┘
+    /// ```
+    /// If the requested epoch points to a null tipset, there are two options:
+    /// Pick the nearest older tipset or pick the nearest younger tipset.
+    /// Requesting epoch 2 with `ResolveNullTipset::TakeYounger` will return
```

Comment:
We could use fancy doc links instead of just marking this as code.

Reply:
You've got it, mr fancy pants. :)